codalab/worker/main.py

```diff
@@ -116,6 +117,11 @@ def parse_args():
         action='store_true',
         help='If specified the worker quits if it finds itself with no jobs after a checkin',
     )
+    parser.add_argument(
+        '--singularity',
```

Comment:
Maybe make an argument called `--container_runtime` which defaults to "docker" and can be set to "singularity".